crates/kitsune_p2p/kitsune_p2p/src/types/event.rs

```diff
@@ -197,17 +197,37 @@ pub enum MetricQueryAnswer {
 
 /// An exclusive range of timestamps, measured in microseconds
 pub type TimeWindow = std::ops::Range<Timestamp>;
+
 /// An inclusive range of timestamps, measured in microseconds
 pub type TimeWindowInclusive = std::ops::RangeInclusive<Timestamp>;
 
 /// A time window which covers all of recordable time
 pub fn full_time_window() -> TimeWindow {
     Timestamp::MIN..Timestamp::MAX
 }
+
 /// A time window which inclusively covers all of recordable time
 pub fn full_time_window_inclusive() -> TimeWindowInclusive {
     Timestamp::MIN..=Timestamp::MAX
 }
+
+/// Generic Kitsune Request of the implementor
+pub enum KGenReq {
+    /// Extrapolated Peer Coverage
+    PeerExtrapCov {
+        /// The space to extrapolate coverage
+        space: Arc<super::KitsuneSpace>,
+        /// Storage arcs of joined agents
+        dht_arc_set: DhtArcSet,
+    },
+}
```

Comment:
Did you do it this way expecting that anything we add in the future can go here to reduce boilerplate? Maybe add a comment if so.

Reply:
```suggestion
/// Generic Kitsune Request of the implementor
/// This enum may be easier to add variants to for future updates,
/// rather than adding a full new top-level event message type.
pub enum KGenReq {
    /// Extrapolated Peer Coverage
    PeerExtrapCov {
        /// The space to extrapolate coverage
        space: Arc<super::KitsuneSpace>,
        /// Storage arcs of joined agents
        dht_arc_set: DhtArcSet,
    },
}
```